force-app/main/default/classes/DMLOperations.cls

```diff
@@ -212,8 +350,23 @@ public with sharing class DMLOperations {
      * @param numOfCases The number of cases to create.
      */
     public static void createAndDeleteCases(String accountId, Integer numOfCases) {
+         // List to hold the Case records
+        List<Case> caseList = new List<case>();
+        
+        // Create the specified number of cases
+        for (Integer i = 0; i < numOfCases; i++) {
+            Case caseToInsert = new Case(AccountId = accountId, Subject = 'Case ' + i);
+            
+            caseList.add(caseToInsert);
+        }
+        
         // Insert the cases
-
+        if (!caseList.isEmpty()) {
+            insert caseList;
+        }
+        
         // Delete the cases
+        if (!caseList.isEmpty()) {
+            delete caseList;
+        }
```

Comment:
It was a best practice in the past to check if a list have values before your DML but it's not needed anymore. Apex is smart enough now to not do any changes if it's empty.